core/dbt/task/deps.py

```diff
@@ -89,16 +89,14 @@ def _create_packages_yml_entry(package: str, version: Optional[str], source: str
 
 class DepsTask(BaseTask):
     def __init__(self, args: Any, project: Project) -> None:
+        project.project_root = str(Path(project.project_root).resolve())
         super().__init__(args=args)
         # N.B. This is a temporary fix for a bug when using relative paths via
         # --project-dir with deps.  A larger overhaul of our path handling methods
         # is needed to fix this the "right" way.
         # See GH-7615
-        project.project_root = str(Path(project.project_root).resolve())
-        self.project = project
-
-        move_to_nearest_project_dir(project.project_root)
         self.cli_vars = args.vars
+        self.project = project
```

Comment:
@rariyama Did you happen to check if preserving [this order](https://github.com/dbt-labs/dbt-core/blob/84eb0ff6720ec82ce4015c2657d512bf51381732/core/dbt/task/deps.py#L92-L101) worked or not?

I'm wondering if you changed the ordering intentionally or if it was just an accidental side-effect when resolving merge conflicts.

i.e., I'm curious if this would work (which is basically identical to [here](https://github.com/dbt-labs/dbt-core/blob/84eb0ff6720ec82ce4015c2657d512bf51381732/core/dbt/task/deps.py#L92-L101) except for removing the `move_to_nearest_project_dir(project.project_root)` line):

```python
        super().__init__(args=args)

        # N.B. This is a temporary fix for a bug when using relative paths via
        # --project-dir with deps.  A larger overhaul of our path handling methods
        # is needed to fix this the "right" way.
        # See GH-7615
        project.project_root = str(Path(project.project_root).resolve())
        self.project = project

        self.cli_vars = args.vars
```

Reply:
I fixed the order of variables as you wrote in this commit and it worked fine.
https://github.com/dbt-labs/dbt-core/pull/9596/commits/4191735ea2c32f83ed4c1ecc1f6816d1b462ea20